include/seqan3/io/alignment_file/input_format_concept.hpp

```diff
@@ -31,11 +31,82 @@
 namespace seqan3::detail
 {
 
+/*!\brief Internal class used to expose the actual format interface to read alignment records from the file.
+ * \ingroup alignment_file
+ *
+ * \tparam format_type The type of the format to be exposed.
+ *
+ * \details
+ *
+ * Exposes the protected member function `read_alignment_record` from the given `format_type`, such that the file can
+ * call the proper function for the selected format.
+ */
 template <typename format_type>
 struct alignment_file_input_format_exposer : public format_type
 {
 public:
-    using format_type::read_alignment_record;
+    // Can't use `using format_type::read_alignment_record` as it produces a hard failure in the format concept check
+    // for types that do not model the format concept, i.e. don't offer the proper read_alignment_record interface.
+    //!\copydoc seqan3::alignment_file_input_format::read_alignment_record
+    template <typename stream_type,
+              typename seq_legal_alph_type,
+              typename ref_seqs_type,
+              typename ref_ids_type,
+              typename seq_type,
+              typename id_type,
+              typename offset_type,
+              typename ref_seq_type,
+              typename ref_id_type,
+              typename ref_offset_type,
+              typename align_type,
+              typename cigar_type,
+              typename flag_type,
+              typename mapq_type,
+              typename qual_type,
+              typename mate_type,
+              typename tag_dict_type,
+              typename e_value_type,
+              typename bit_score_type>
+    void read_alignment_record(stream_type & stream,
```

Comment:
Can you replace this at least with

```cpp
template <typename arg_types...>
void read_alignment_record(arg_types ...args)
{ // ...
}
``` 
? We do enough concept checks to enforce the proper parameters?